File: lib/veranstaltung.cpp
#include <veranstaltung.h>
#include <cxxtools/serializationinfo.h>
#include <cxxtools/utf8.h>

void operator>>= (const cxxtools::SerializationInfo& si, Veranstaltung& v)
{
  si.getMember("vid") >>= v.vid;
  si.getMember("name") >>= cxxtools::Utf8(v.name);
  si.getMember("datum") >>= v.datum;
  si.getMember("ort") >>= cxxtools::Utf8(v.ort);
  si.getMember("logo") >>= cxxtools::Utf8(v.logo);
}

void operator<<= (cxxtools::SerializationInfo& si, const Veranstaltung& v)
{
  si.setTypeName("Veranstaltung");

  si.addMember("vid") <<= v.vid;
  si.addMember("name") <<= cxxtools::Utf8(v.name);
  si.addMember("datum") <<= v.datum;
  si.addMember("ort") <<= cxxtools::Utf8(v.ort);
  si.addMember("logo") <<= cxxtools::Utf8(v.logo);
}

File: resources/veranstaltung.js
function reload()
{
    window.location.reload();
}

$('input[name="vid"]').click(function() {
    $.getJSON('/setveranstaltung.json',
    {
        "vid": this.value
    },
    function (reply) {
        document.title = reply.name;
        information("Veranstaltung <i>" + reply.name + "</i> ausgewählt");
    })
});

var veranstaltungenTable = $('#veranstaltungen').DataTable({
    ajax: {
        url: 'veranstaltung.json',
        dataSrc: ''
    },
    order: [0, 'desc'],
    select: true,
    columns: [
        { data: 'vid' },
        { data: 'name' },
        { data: 'datum' },
        { data: 'ort' },
        { data: 'logo' }
      ]
});

veranstaltungenTable.on('select', function (e, dt, type, indexes) {
    console.log(veranstaltungenTable.rows(indexes).data()[0].vid);
    tntnet.action('setveranstaltung', {
        vid: veranstaltungenTable.rows(indexes).data()[0].vid
    });
});

var initdialog = function() {
        $('label').css('width', '100px');
        var dialog = $(this).dialog({
          height: 300,
          width: 400,
          modal: true,
          buttons: {
            "Speichern": function() {
              tntnet.action('veranstaltung/save', {
                  vid: $('#vid').val(),
                  name: $('#name').val(),
                  ort: $('#ort').val(),
                  datum: $('#datum').val(),
                  logo: $('#logo').val()
              },
              function () {
                  information("Veranstaltung gespeichert");
                  dialog.dialog("close");
                  reload();
              })
            },
            "Abbrechen": function() {
              dialog.dialog("close");
            }
          }
        });
        $('.datum').datepicker();
      };

$('#neu').click(function() {
  $('#dialog')
    .load('veranstaltung.html', { neu: true }, initdialog)
});

$('#bearbeiten').click(function() {
  $('#dialog')
    .load('veranstaltung.html', initdialog)
});

$('#loeschen').click(function() {
    var dialog = $('#dialog-confirm').dialog({
        resizable: false,
        height: 240,
        width: 400,
        modal: true,
        buttons: {
          "Veranstaltung löschen": function() {
            tntnet.action('veranstaltung/del', {
            },
            function () {
                information("Veranstaltung gelöscht");
                dialog.dialog( "close" );
                reload();
            })
          },
          "Abbrechen": function() {
            $( this ).dialog( "close" );
          }
        }
    });
});
